games/achievements.py

```diff
@@ -42,101 +55,150 @@ class DNFAchievement(Achievement):
     description = "Participated in a game that completed, where you didn't"
     icon = "coffin.svg"
 
-    def has_achieved(user):
-        return (
+    def get_level(user):
+        if (
             user.gameplayer_set.filter(dnf=True)
             .filter(game__dnf=False, game__end_datetime__isnull=False)
             .exists()
-        )
+        ):
+            return AchievementLevel.GOLD
+        return AchievementLevel.NO_LEVEL
 
 
-class Top10Achievement(Achievement):
-    name = "Top 10"
-    description = "Placed top 10 total sips in a season"
+class TopAchievement(Achievement):
+    name = "Top"
+    description = "Placed top (10/5/3/1) total sips in a season"
     icon = "trophy-cup.svg"
 
-    def has_achieved(user):
+    def get_level(user):
         current_season = Season.current_season()
+        highest_rank = 99
+
         for i in range(1, current_season.number):  # Exclude current season
-            top10 = (
+            top10 = list(
                 PlayerStat.objects.filter(season_number=i)
                 .order_by("-total_sips")[:10]
                 .values("user")
             )
-            if {"user": user.id} in top10:
-                return True
-
-        return False
+            try:
+                rank = top10.index({"user": user.id})
+                highest_rank = min(highest_rank, rank)
```

Comment:
Reverse the `__lt__` implementation of `AchievementLevel` so that this can use `max` instead, which is more natural.